src/base/bittorrent/torrentcontentremover.cpp

```diff
@@ -0,0 +1,61 @@
+/*
+ * Bittorrent Client using Qt and libtorrent.
+ * Copyright (C) 2024  Vladimir Golovnev <glassez@yandex.ru>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#include "torrentcontentremover.h"
+
+#include "base/utils/fs.h"
+
+void BitTorrent::TorrentContentRemover::performJob(const QString &torrentName, const Path &basePath
+        , const PathList &fileNames, TorrentContentRemoveOption option)
+{
+    QString errorMessage;
+
+    if (!fileNames.isEmpty())
+    {
+        const auto removeFileFn = [&option](const Path &filePath)
+        {
+            return ((option == TorrentContentRemoveOption::MoveToTrash)
+                    ? Utils::Fs::moveFileToTrash : Utils::Fs::removeFile)(filePath);
+        };
+
+        for (const Path &fileName : fileNames)
+        {
+            if (const auto result = removeFileFn(basePath / fileName)
+                    ; !result && errorMessage.isEmpty())
```

Comment:
Wouldn't it be better if the error messages were collected as a list?
And each file correlates to its own error message.

Reply:
> Wouldn't it be better if the error messages were collected as a list? And each file correlates to its own error message.

https://github.com/qbittorrent/qBittorrent/pull/20252#discussion_r1447612674

Reply:
Hypothetically, some files might be locked by some process (user opened it and forgot to close it) and other files might have other issues (maybe user meddled with permissions). I don't think following libtorrent one-error behavior is preferable here, especially the error message is from Qt.
I don't strongly insist but I consider the current error reporting resolution is not enough.

Reply:
> I consider the current error reporting resolution is not enough.

Since it is no worse than what was previously (provided by libtorrent), I do not consider myself required to improve it.
Personally, I don't think it's a good idea to report every file to the log, and I can't imagine any other purpose for collecting these errors.
As for the log, I think that in the vast majority of cases it is enough just to see that something went wrong when removing the content of torrent. It is not supposed that the user looks in the log immediately after each action, and when he does notice it, the reasons for the errors may no longer be relevant. BTW, in this case, a pop-up message could be more useful, but again, you won't put a detailed report on all the files there.
Well, as always, further improvements are welcome.

Reply:
>As for the log, I think that in the vast majority of cases it is enough just to see that something went wrong when removing the content of torrent. It is not supposed that the user looks in the log immediately after each action, and when he does notice it, the reasons for the errors may no longer be relevant.

Just for discussion.
It helps when a user is doing post mortem about why some files fail to remove. Without it, it wouldn't be possible to learn precisely what the culprit was when the action performed (via automated rules) and avoid it next time.

Reply:
The problem with torrents containing thousands of files, the log will just be filled with the same type of messages, and even if for some files the reasons will be different, it will be very difficult to find it from the log.

Reply:
OK. Lets (me) not worry about it for now.